airbyte-db/src/main/java/io/airbyte/db/JdbcUtils.java

```diff
@@ -0,0 +1,231 @@
+        final Timestamp t = r.getTimestamp(i);
+        java.util.Date d = new java.util.Date(t.getTime() + (t.getNanos() / 1000000));
+        o.put(columnName, toISO8601String(d));
+      }
+      case BINARY, VARBINARY, LONGVARBINARY -> o.put(columnName, r.getBytes(i));
+      default -> o.put(columnName, r.getString(i));
+    }
+  }
+
+  private static String toISO8601String(java.util.Date date) {
+    return DATE_FORMAT.format(date);
+  }
+
+  public static void setFieldWithType(PreparedStatement preparedStatement,
+                                      int parameterIndex,
+                                      JDBCType cursorFieldType,
+                                      String cursor)
+      throws SQLException {
+    switch (cursorFieldType) {
+      // parse date, time, and timestamp the same way. this seems to not cause an problems and allows us
+      // to treat them all as ISO8601. if this causes any problems down the line, we can adjust.
+      case DATE, TIME, TIMESTAMP -> {
+        try {
+          preparedStatement.setTimestamp(1, Timestamp.from(DATE_FORMAT.parse(cursor).toInstant()));
+        } catch (ParseException e) {
+          throw new RuntimeException(e);
+        }
+      }
+      case TINYINT, SMALLINT -> preparedStatement.setShort(parameterIndex, Short.parseShort(cursor));
+      case INTEGER -> preparedStatement.setInt(parameterIndex, Integer.parseInt(cursor));
+      case BIGINT -> preparedStatement.setLong(parameterIndex, Long.parseLong(cursor));
+      case FLOAT, DOUBLE, REAL -> preparedStatement.setDouble(parameterIndex, Double.parseDouble(cursor));
+      case NUMERIC, DECIMAL -> preparedStatement.setBigDecimal(parameterIndex, new BigDecimal(cursor));
+      case CHAR, VARCHAR, LONGVARCHAR -> preparedStatement.setString(parameterIndex, cursor);
+      default -> throw new IllegalArgumentException(String.format("%s is not supported for incremental.", cursorFieldType));
+    }
+  }
+
+  // the switch statement intentionally has duplicates so that its structure matches the type switch
+  // statement above.
+  @SuppressWarnings("DuplicateBranchesInSwitch")
+  public static JsonSchemaPrimitive getType(JDBCType jdbcType) {
+    return switch (jdbcType) {
+      case BIT, BOOLEAN -> JsonSchemaPrimitive.BOOLEAN;
+      case TINYINT, SMALLINT -> JsonSchemaPrimitive.NUMBER;
+      case INTEGER -> JsonSchemaPrimitive.NUMBER;
+      case BIGINT -> JsonSchemaPrimitive.NUMBER;
+      case FLOAT, DOUBLE -> JsonSchemaPrimitive.NUMBER;
+      case REAL -> JsonSchemaPrimitive.NUMBER;
+      case NUMERIC, DECIMAL -> JsonSchemaPrimitive.NUMBER;
+      case CHAR, VARCHAR, LONGVARCHAR -> JsonSchemaPrimitive.STRING;
+      case DATE -> JsonSchemaPrimitive.STRING;
+      case TIME -> JsonSchemaPrimitive.STRING;
+      case TIMESTAMP -> JsonSchemaPrimitive.STRING;
+      case BINARY, VARBINARY, LONGVARBINARY -> JsonSchemaPrimitive.STRING;
```

Comment:
is that correct?

Reply:
what did you have in mind? we can represent an arbitrarily long byte array as a string in hex. given the data type primitives we have at our disposal this seems like the best fit.